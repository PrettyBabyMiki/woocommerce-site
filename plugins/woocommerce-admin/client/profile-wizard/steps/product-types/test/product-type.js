/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import userEvent from '@testing-library/user-event';

/**
 * Internal dependencies
 */
import ProductType from '../product-type';

const defaultProps = {
	annualPrice: 120,
	label: 'Product type label',
	description: 'Product type description',
	moreUrl: 'https://woocommerce.com/my-product-type',
	slug: 'my-product-type',
};

describe( 'ProductType', () => {
	test( 'should render the product type', () => {
		const productType = shallow(
			<ProductType { ...defaultProps } isMonthlyPricing={ false } />
		);
		expect( productType ).toMatchSnapshot();
	} );

	test( 'should render the product type with monthly prices', () => {
		const productType = shallow(
			<ProductType { ...defaultProps } isMonthlyPricing={ true } />
		);
		expect( productType ).toMatchSnapshot();
	} );

	test( 'should show Popover on click', () => {
		const { container } = render(
			<ProductType { ...defaultProps } isMonthlyPricing={ true } />
		);

		const infoButton = screen.getByLabelText(
			'Learn more about recommended free business features',
			{
				selector: 'button',
			}
		);

		userEvent.click( infoButton );

		const popover = container.querySelector( '.components-popover' );
		const learnMoreLink = popover.querySelector( 'a' );
		expect( popover ).not.toBeNull();
		expect( popover.textContent ).toBe(
			defaultProps.description + ' Learn more'
		);
		expect( learnMoreLink.href ).toBe( defaultProps.moreUrl );
	} );
} );
