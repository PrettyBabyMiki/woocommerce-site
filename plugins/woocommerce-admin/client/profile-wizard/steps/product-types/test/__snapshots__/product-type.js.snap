// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductType should render the product type 1`] = `
<div
  className="woocommerce-product-type"
>
  <span
    className="woocommerce-product-type__label"
  >
    Product type label
  </span>
  <ForwardRef(Button)
    isTertiary={true}
    label="Learn more about recommended free business features"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="material-icons-outlined"
    >
      info
    </i>
  </ForwardRef(Button)>
  <Tooltip
    position="bottom center"
    text="This product type requires a paid extension.
We'll add this to a cart so that
you can purchase and install it later."
  >
    <Pill>
      <span
        className="screen-reader-text"
      >
        This product type requires a paid extension.
We'll add this to a cart so that
you can purchase and install it later.
      </span>
      $120 per year
    </Pill>
  </Tooltip>
</div>
`;

exports[`ProductType should render the product type with monthly prices 1`] = `
<div
  className="woocommerce-product-type"
>
  <span
    className="woocommerce-product-type__label"
  >
    Product type label
  </span>
  <ForwardRef(Button)
    isTertiary={true}
    label="Learn more about recommended free business features"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="material-icons-outlined"
    >
      info
    </i>
  </ForwardRef(Button)>
  <Tooltip
    position="bottom center"
    text="This product type requires a paid extension.
We'll add this to a cart so that
you can purchase and install it later."
  >
    <Pill>
      <span
        className="screen-reader-text"
      >
        This product type requires a paid extension.
We'll add this to a cart so that
you can purchase and install it later.
      </span>
      $10 per month
    </Pill>
  </Tooltip>
</div>
`;
